src/modules/configurator/components/Discount.tsx

```diff
@@ -1,11 +1,52 @@
-import React from "react";
+import React, { useState } from 'react';
+import { useResetRecoilState, useSetRecoilState } from 'recoil';
+import { discountValueState } from '../state';
 
 export const Discount: React.FC = () => {
+  const [discountInput, setdiscountInput] = useState<number | string>('');
+  const setDiscountValue = useSetRecoilState(discountValueState);
+  const resetDiscount = useResetRecoilState(discountValueState);
+
+  const handleDiscountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const discount = parseInt(e.target.value);
+    switch (true) {
+      case isNaN(discount):
+        setdiscountInput('');
+        break;
+      case discount > 100:
+        setdiscountInput(100);
+        break;
+      case discount > 0:
+        setdiscountInput(discount);
+        break;
+      default:
+        setdiscountInput(0);
+    }
+  };
+
+  const handleApplyDiscount = () => {
```

Comment:
Also here. When you use the ternary operator, you want the track what is changing.
```
const discountValue = typeof discountInput === 'string' ? 0 : discountInput;
setDiscountValue(dicscountValue)
```
The function is the same, only the value changes.